onboarding/source/amb_light_service.c

```diff
@@ -26,14 +59,60 @@ uint8_t initLightService(void) {
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+
+    light_event_t event;
+    while(1) {
+        if (xQueueReceive(eventQueue, &event, (TickType_t) 0) == pdPASS) {
+            if(event == MEASURE_LIGHT) {
+                // if event recieved is MEASURE_LIGHT, get value from adc and send over serial
+                uint16_t data = getAmbientLightData(); // CHANGE: data now uint16_t since ADC is 12-bit
+                unsigned char adc_value[10]; 
+                sprintf(adc_value, "%d", data); // CHANGE: convert ADC integer value to char
```

Comment:
For future reference, the `snprintf` function is more secure than `sprintf` and should be preferred.
 
The problem with `sprintf` is that it's possible to write more data to the `adc_value` buffer than the size of the buffer itself. `snprintf` makes sure you don't exceed the limit of the buffer. 

However, in this case, your buffer size is 10 and `data` should never be more than 5 digits. So it will still work.